wicket/src/cli/rack_update.rs

```diff
@@ -0,0 +1,313 @@
+    /// Attach to one or more running updates.
+    Attach(AttachArgs),
+}
+
+impl RackUpdateArgs {
+    pub(crate) fn exec(
+        self,
+        log: Logger,
+        wicketd_addr: SocketAddrV6,
+        global_opts: GlobalOpts,
+    ) -> Result<()> {
+        let runtime =
+            tokio::runtime::Runtime::new().context("creating tokio runtime")?;
+        runtime.block_on(self.exec_impl(log, wicketd_addr, global_opts))
+    }
+
+    async fn exec_impl(
+        self,
+        log: Logger,
+        wicketd_addr: SocketAddrV6,
+        global_opts: GlobalOpts,
+    ) -> Result<()> {
+        match self {
+            RackUpdateArgs::Start(args) => {
+                args.exec(log, wicketd_addr, global_opts).await
+            }
+            RackUpdateArgs::Attach(args) => {
+                args.exec(log, wicketd_addr, global_opts).await
+            }
+        }
+    }
+}
+
+#[derive(Debug, Args)]
+pub(crate) struct StartRackUpdateArgs {
+    #[clap(flatten)]
+    component_ids: ComponentIdSelector,
+
+    /// Force update the RoT even if the version is the same.
+    #[clap(long, help_heading = "Update options")]
+    force_update_rot: bool,
+
+    /// Force update the SP even if the version is the same.
+    #[clap(long, help_heading = "Update options")]
+    force_update_sp: bool,
+
+    /// Detach after starting the update.
+    ///
+    /// The `attach` command can be used to reattach to the running update.
+    #[clap(short, long, help_heading = "Update options")]
+    detach: bool,
+}
+
+impl StartRackUpdateArgs {
+    async fn exec(
+        self,
+        log: Logger,
+        wicketd_addr: SocketAddrV6,
+        global_opts: GlobalOpts,
+    ) -> Result<()> {
+        let client = create_wicketd_client(&log, wicketd_addr, WICKETD_TIMEOUT);
+
+        let update_ids = self.component_ids.to_component_ids()?;
+        let options = CreateStartUpdateOptions {
+            force_update_rot: self.force_update_rot,
+            force_update_sp: self.force_update_sp,
+        }
+        .to_start_update_options()?;
+
+        let num_update_ids = update_ids.len();
+
+        let params = StartUpdateParams {
+            targets: update_ids.iter().copied().map(Into::into).collect(),
+            options,
+        };
+
+        slog::debug!(log, "Sending post_start_update"; "num_update_ids" => num_update_ids);
+        match client.post_start_update(&params).await {
+            Ok(_) => {
+                slog::info!(log, "Update started for {num_update_ids} targets");
+            }
+            Err(error) => {
+                // Error responses can be printed out more clearly.
+                if let wicketd_client::Error::ErrorResponse(rv) = &error {
+                    slog::error!(
+                        log,
+                        "Error response from wicketd: {}",
+                        rv.message
+                    );
+                    bail!("Received error from wicketd while starting update");
```

Comment:
Should we include `rv.message` in this `bail!`? Or is `log` visible to the caller and not logging to `/tmp` or wherever the default wicket log ends up?

Reply:
For interactive use, things are logged to both stderr and the wicket log file -- so this is fine.